src/options.rs

```diff
@@ -122,18 +157,42 @@ impl Options {
 	}
 
 	pub fn load_and_validate_metadata(&self) -> Result<Option<Salt>> {
+		let mut path: PathBuf = self.path.clone();
+		path.push("metadata");
+		let (cols, mut salt) = Self::load_metadata(&path)?;
+
+		if let Some(mut cols) = cols {
+			if cols.len() != self.columns.len() {
+				return Err(Error::InvalidConfiguration("Column config mismatch".into()));
+			}
+
+			for c in 0..cols.len() {
+				cols[c].compression_treshold = self.columns[c].compression_treshold;
+				if cols[c] != self.columns[c] {
+					return Err(Error::InvalidConfiguration(format!(
+								"Column config mismatch for column {}. Expected \"{}\", got \"{}\"",
+								c, self.columns[c].as_string(), cols[c].as_string())));
+				}
+			}
+		} else {
+			let s: Salt = self.salt.unwrap_or(rand::thread_rng().gen());
```

Comment:
So with these changes we always be using salt?

Reply:
This just allows overriding the salt in the options, but does not change the logic. The salt won't be set for an existing metadata that does not already has it.